scripts/gas-refund-program/transactions-indexing/successful-swap-tx-indexing.ts

```diff
@@ -111,7 +107,9 @@ export async function computeSuccessfulSwapsTxFeesRefund({
       `fetched ${swaps.length} swaps between ${_startTimestampSlice} and ${_endTimestampSlice}`,
     );
 
-    const updatedPendingGasRefundDataByAddress: TxFeesByAddress = {};
+    const pendingGasRefundTransactionData: GasRefundTransactionData[] = [];
+    // hash list to track if we've seen a tx already (just checks against timeslices, we assume tx duplicates have the same timestamp)
+    let txsSeen: Record<string, boolean> = {}
```

Comment:
What I imagined was one single preprocessing step : apply _.uniqBy(‘hash’) to swaps array (conditionally depending on epoch).  I think it’s less verbose and less error

Reply:
Actually it can be a post processing step of getSuccessfulSwaps (inside that function)

That way we can also put an assert to make sure this never happens.

basically it will be 
- fetch swaps
- if epoch < GasRefundTxOriginCheckStartEpoch, return swaps
- produce new array swapsWithTxOriginEqInitiator
- produce new array uniqSwaps = uniqBy(swapsWithTxOriginEqInitiator, ‘txHash’)
- assert(uniqSwaps.length === swapsWithTxOriginEqInitiator.length, ‘duplicates found’)
- return uniqSwaps